engine/src/main/java/org/terasology/engine/modes/loadProcesses/PostBeginSystems.java

```diff
@@ -41,7 +47,12 @@ public String getMessage() {
     @Override
     public boolean step() {
         if (componentSystems.hasNext()) {
-            componentSystems.next().postBegin();
+            try {
+                currentSystem = componentSystems.next();
+                currentSystem.postBegin();
+            } catch(Exception e){
```

Comment:
You should be more specific about the Exception caught here. I think NoSuchElement was the cause of the error, so this one should be caught here.